api/models/model_class_possibilities/Model_id_recursion.py

```diff
@@ -0,0 +1,60 @@
+# for more detail, see class Model in :
+# https://github.com/pallets/flask-sqlalchemy/blob/master/flask_sqlalchemy/model.py
+from typing import Any, List, Type
+
+
+class MyModel(object):
+    query_class = None  # flask_alchemy attribute
+    query = None  # flask_alchemy attribute
+
+    DONOTSEND_MODEL = {'_sa_instance_state'}
+    DONOTSEND = []
+    _jsonified = None
+
+    def __repr__(self) -> str:
+        return '<{}>'.format(self.__class__.__name__)
+
+    def to_dict_recursive(self) -> dict:
+        return self._to_dict_recursive(list_objects_id_passed_through=[id(self)])
+
+    def _to_dict_recursive(self, list_objects_id_passed_through: List[int]) -> dict:
+        # functions :
+        # anti_circular_recursion : check if we've already called the object
+        #                               if not do the recursion
+        # type_shunt_recursive : select the actions for each type of attr
+
+        def anti_circular_recursion(obj: Type[MyModel]) -> any:
```

Comment:
maybe naming this differently to its functionality because having `anti` confused me

Reply:
I don't have a clue how to call it ? `def recursive(...):` ?